### PR TITLE
Feature: Themes

### PR DESCRIPTION
**Pull request by chriscoyier**
The idea is you do a command like:

```
:theme original
:theme dark-king
```

And switch between themes.

### CODE REVIEW

**Comment by chriscoyier** [comment]
This also represents the proper Sassification of the project. So once this is cleaned up, I hope we can get it into master fairly soon. Shouldn't affect other features very much I hope!

**Comment by chriscoyier** [comment]
This is ready for a look

File: src/vimish/index.js [hunk 1]
```diff
@@ -322,15 +323,22 @@ exports.stream = exports.streams = Command.of(
...
-        state == 1 ? "on" : "off"
-      }`,
-    })
+    Bus.emit(Bus.events.REDRAW)
   }
 )
 
+// `:theme dark-king`
+exports.theme = Command.of(["value"], async ({ value }) => {
+  if (value in THEME_NAMES) {
```
Comment (ondreian) [inline src/vimish/index.js:332]: This should be a return early/error first pattern.

basically anytime you write an if/else statement and you have a tiny "dangling else" it should be refactored to be:

```javascript
if (!(value in THEME_NAMES)) {
  // handle dangling case first
}
```
This prevents complexity from building up over time as more cases are added, and makes it easier to verify what is required for the `Ok` path to be run.

Here is a pretty solid overview of else-less programming paradigm if you're interested
https://blog.timoxley.com/post/47041269194/avoid-else-return-early

**Comment by ondreian** [comment]
The feature seems cool, I think eventually we might want some sort of `:theme download <name> <url>` so people can put them online and share them or whatever and this sort of theme changing logic will be very nice for that.

File: src/vimish/index.js (hunk 1 above)
Comment (chriscoyier) [reply src/vimish/index.js]: Can-do.

**Comment by chriscoyier** [comment]
Yeah downloading themes would be great. The `user.css` feature is already cool for quick sharing of customizations, but the people who would do that are 1% of the people who would type in a command to get a sweet new theme.

**Pull request merged**